src/frontend/AppContainerWrapper.tsx

```diff
@@ -154,17 +155,19 @@ const AppContainerWrapper = () => {
   }, [inviteModalEnabled, queuedInvite, openInviteModal]);
 
   return (
-    <AppContainer
-      loadNavigationState={loadNavigationState}
-      onNavigationStateChange={onNavStateChange}
-      persistNavigationState={persistNavigationState}
-      ref={nav => {
-        if (nav) {
-          navRef.current = nav;
-        }
-      }}
-      uriPrefix={URI_PREFIX}
-    />
+    <PracticeMode>
```

Comment:
I would update this component to take a prop e.g. `hideBar` that's determined based on the nav state. then we can set some app-level state variable that calculates the passed value whenever the navigation state changes i.e.

```tsx
const [hidePracticeModeBar, setHidePracticeModeBar] = React.useState(() => {
  // do an initial calculation based on the config
  // or whatever we use to determine if the we start out in practice mode
  ...
}

const onNavigationStateChange = (state) => {
  ...
  // Figure out if the screen is a screen where we want to hide practice mode UI
  const hideBar = someHelperFunction(state)
  setHidePracticeModeBar(hideBar)
  ... 
}

return (
  {/* Use the prop in this component to conditionally display the bar */}
  <PracticeMode hideBar={hidePracticeModeBar} >
    <AppContainer onNavigationStateChange={onNavigationStateChange}  ... />
  </PracticeMode>
)
```

With this, you can remove the added `practiceModeBar` and `practiceModeBarToggle` fields in the config context as you currently have it and not worry about having to update individuals components/screens to call something when they mount to handle this.

Reply:
yeah, this architecture is probably more maintainable than my current architecture